azurecontainerapps.ts

```diff
@@ -343,16 +349,17 @@ export class azurecontainerapps {
         // If telemetry is enabled, log that the previously built image scenario was targeted for this task
         this.telemetryHelper.setImageScenario();
     }
-
     /**
-     * Builds a runnable application image using a Dockerfile or the builder and pushes it to the Container Registry.
-     */
-    private static async buildAndPushImageAsync() {
+        * Sets up the Container App Image properties if it's not provided by the user.
+        * file is provided.
+        */
+    private static setupContainerAppImageProperties() {
         // Get the name of the image to build if it was provided, or generate it from build variables
         this.imageToBuild = this.toolHelper.getInput('imageToBuild', false);
+
         if (this.util.isNullOrEmpty(this.imageToBuild)) {
             const imageRepository = this.toolHelper.getDefaultImageRepository()
-            this.imageToBuild = `${this.registryUrl}/${imageRepository}:${this.buildId}.${this.buildNumber}`;
+            this.imageToBuild = this.util.isNullOrEmpty(this.registryUrl) ? `default/${imageRepository}:${this.buildId}.${this.buildNumber}` : `${this.registryUrl}/${imageRepository}:${this.buildId}.${this.buildNumber}`;
```

Comment:
Do we use this `imageToBuild` value anywhere for the internal registry scenario? I haven't checked everywhere yet, but it seems like the main thing we use it for is to set the value of `useInternalRegistry`, but it seems like for that case we really are just checking if `this.registryUrl` is null or empty (since we set this `imageToBuild` value to `default/*` if no registry was provided by the user)

Reply:
That is correct! But for update scenarios, I understand the CLI may have context of the existing container app's registry info but the action would not?
so, we need this `imageToBuild` arg to determine if we want to add `--source` to the `az containerapp update` command?